sources/PTYTextView.m

```diff
@@ -1434,7 +1435,8 @@ - (void)keyDown:(NSEvent*)event {
         // track the instance of PTYTextView that is currently handling a key event and rerouting
         // calls as needed in -insertText and -doCommandBySelector.
         gCurrentKeyEventTextView = [[self retain] autorelease];
-        [self interpretKeyEvents:[NSArray arrayWithObject:event]];
+        _keydownEvent = event;
+        _inputMethodIsInserting = !event.isARepeat && [self.inputContext handleEvent:event];
```

Comment:
I disassembled `-[NSView interpretKeyEvents:]` on the 10.11 SDK and here's what it does:

````
- (void)interpretKeyEvents:(NSArray *)events {
  NSTextInputContext *inputContext = [self inputContext];
  if (!inputContext) {
    [super interpretKeyEvents:events];
    return;
  }
  for (NSEvent *event in events) {
    [inputContext handleEvent:event];
  }
}
````

(The annotated disassembly is here: https://gist.github.com/gnachman/f827f205e8b99fe35c9b5fbd2235aa28)

Are you absolutely sure that it's the call to handleEvent: that causes issues with key repeat?